config/v1/types_cloudprovider.go

```diff
@@ -0,0 +1,37 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CloudProvider holds cluster-wide information about CloudProvider.  The canonical name is `cluster`
+type CloudProvider struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	Spec CloudProviderSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	Status CloudProviderStatus `json:"status"`
+}
+
+type CloudProviderSpec struct {
```

Comment:
The lowest common denominator I know of here would be:


```go
Name string `json:"name"`
```

Where `Name` maps to the kube-controller-manager `--cloud-provider` argument.

Reply:
Or `Provider`.